perma_web/api2/views.py

```diff
@@ -0,0 +1,483 @@
+
+# /capture_jobs
+class CaptureJobListView(BaseView):
+    serializer_class = CaptureJobSerializer
+
+    def get(self, request, format=None):
+        """ List capture_jobs for user. """
+        queryset = CaptureJob.objects.filter(link__created_by_id=request.user.pk, status__in=['pending', 'in_progress'])
+        return self.simple_list(request, queryset)
+
+
+# /capture_jobs/:id
+# /capture_jobs/:guid
+class CaptureJobDetailView(BaseView):
+    serializer_class = CaptureJobSerializer
+
+    def get(self, request, pk=None, guid=None, format=None):
+        """ Single capture_job details. """
+        if guid:
+            # We were called as /capture_jobs/:guid
+            # Return capture_job for given link_id
+            obj = self.get_object_for_user(request.user, CaptureJob.objects.filter(link_id=guid).select_related('link'))
+            return self.simple_get(request, obj=obj)
+        else:
+            # We were called as /capture_jobs/:id
+            return self.simple_get(request, pk)
+
+
+### LINK VIEWS ###
+
+class LinkFilter(django_filters.rest_framework.FilterSet):
+    """
+        Custom filter for searching links by query string.
+    """
+    date = django_filters.IsoDateTimeFilter(name="creation_timestamp", lookup_expr='date')      # ?date=
+    min_date = django_filters.IsoDateTimeFilter(name="creation_timestamp", lookup_expr='gte')   # ?min_date=
+    max_date = django_filters.IsoDateTimeFilter(name="creation_timestamp", lookup_expr='lte')   # ?max_date=
+    url = django_filters.CharFilter(name="submitted_url", lookup_expr='icontains')              # ?url=
+    class Meta:
+        model = Link
+        fields = ['url', 'date', 'min_date', 'max_date']
+
+
+# /public/archives
+class PublicLinkListView(BaseView):
+    permission_classes = ()  # no login required
+    serializer_class = LinkSerializer
+    filter_class = LinkFilter
+    search_fields = ('guid', 'submitted_url', 'submitted_title', 'notes')  # fields that can be searched with q= query string
+
+    def get(self, request, format=None):
+        """ List public links. """
+        queryset = Link.objects.order_by('-creation_timestamp').prefetch_related('captures').discoverable()
+        return self.simple_list(request, queryset)
+
+# /public/archives/:guid
+class PublicLinkDetailView(BaseView):
+    permission_classes = ()  # no login required
+    serializer_class = LinkSerializer
+
+    def get(self, request, guid, format=None):
+        """ Get public link details. """
+        try:
+            obj = Link.objects.discoverable().get(pk=guid)
+        except Link.DoesNotExist:
+            raise Http404
+        return self.simple_get(request, obj=obj)
+
+# /archives
+# /folders/:parent_id/archives
+class AuthenticatedLinkListView(BaseView):
+    serializer_class = AuthenticatedLinkSerializer
+    filter_class = LinkFilter
+    search_fields = PublicLinkListView.search_fields + ('notes',)  # private links can also be searched by notes field
```

Comment:
Hmmm, PublicLinkListView's tuple of search_fields above contains notes. Sounds like it shouldn't.